mcp_server_gravitino/server/tools/models.py

```diff
@@ -0,0 +1,194 @@
+
+
+def get_list_model_versions_by_fqn(mcp: FastMCP, session: httpx.Client) -> None:
+    """List all model versions by fully qualified model name."""
+
+    @mcp.tool(
+        name="get_list_model_versions_by_fqn",
+        description="List all model versions by using fully qualified model name.",
+        tags={
+            "model version",
+            "list operation",
+        },
+    )
+    def _list_model_versions_by_fqn(fqn: str) -> list[dict[str, Any]]:
+        """
+        List all versions of a model by its fully qualified name.
+
+        Parameters
+        ----------
+        fqn : str
+            Fully qualified model name, of the form 'catalog.schema.model'
+            or 'metalake.catalog.schema.model'.
+
+        Returns
+        -------
+        list[dict[str, Any]]
+            A list of versions, each represented as:
+            - version: Version identifier
+        """
+        metalake_name, catalog_name, schema_name, model_name = _parse_four_level_fqn(fqn.split("."))
+        if not metalake_name:
+            metalake_name = global_metalake_name
+
+        response = session.get(
+            f"/api/metalakes/{metalake_name}/catalogs/{catalog_name}/schemas/{schema_name}/models/{model_name}/versions"
+        )
+        response.raise_for_status()
+        response_json = response.json()
+        versions = response_json.get("versions", [])
+
+        version_objects = [
+            _get_model_version_by_fqn_and_version_response(session, fqn, version).get("modelVersion")
+            for version in versions
+        ]
+
+        return [
+            {
+                "version": obj.get("version"),
+                "comment": obj.get("comment"),
+                "aliases": ",".join(obj.get("aliases")),
+                "uri": obj.get("uri"),
+                "creator": obj.get("audit").get("creator"),
+            }
+            for obj in version_objects
+        ]
+
+
+def _get_model_version_by_fqn_and_version_response(session: httpx.Client, fully_qualified_name: str, version: str):
+    """
+    Get a model version by fully qualified model name and version.
+    Parameters
+    ----------
+    session : httpx.Client
+        HTTP client to make requests to Metalake API.
+    fully_qualified_name : str
+        Fully qualified name of the model.
+    version : str
+        Version of the model version.
+    Returns
+    -------
+    dict
+        Response from Metalake API.
+    """
+    model_names = fully_qualified_name.split(".")
+    metalake_name, catalog_name, schema_name, model_name = _parse_four_level_fqn(model_names)
+    if not metalake_name:
+        metalake_name = global_metalake_name
+
+    response = session.get(
+        f"/api/metalakes/{metalake_name}/catalogs/{catalog_name}/schemas/{schema_name}/models/{model_name}/versions/{version}"
+    )
+    response.raise_for_status()
+    return response.json()
+
+
+def _parse_four_level_fqn(fqn: List[str]) -> Tuple[Optional[str], str, str, str]:
```

Comment:
these two methods are common to all tools; if they can be moved to a shared py file, that would be nice!